src/objects/zcl_abapgit_object_aifc.clas.abap

```diff
@@ -0,0 +1,509 @@
+    APPEND zif_abapgit_object=>gc_step_id-abap TO rt_steps.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_metadata.
+    RETURN.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_active.
+    rv_active = abap_false.
+    IF zif_abapgit_object~exists( ) = abap_false.
+      RETURN.
+    ENDIF.
+    rv_active = abap_true.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_locked.
+    rv_is_locked = abap_false.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~jump.
+    TYPES: ty_rsparamsl_255_t TYPE STANDARD TABLE OF rsparamsl_255 WITH NON-UNIQUE DEFAULT KEY.
+
+    DATA lv_report TYPE progname VALUE '/AIF/CONTENT_DISPLAY'.
+    DATA lt_params TYPE ty_rsparamsl_255_t.
+    DATA ls_param LIKE LINE OF lt_params.
+
+    ls_param-selname = 'P_DEPL'.
+    ls_param-kind = 'P'.
+    ls_param-sign = 'I'.
+    ls_param-option = 'EQ'.
+    ls_param-low = ms_icd_data_key-depl_scenario.
+    APPEND ls_param TO lt_params.
+
+    SUBMIT (lv_report) WITH SELECTION-TABLE lt_params AND RETURN.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~serialize.
+    DATA: lx_root TYPE REF TO cx_root.
+    DATA: lx_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+
+    DATA ls_icd_data_key TYPE ty_icd_data_key.
+    DATA lt_ifdata TYPE ty_table_data_t.
+
+    DATA lr_data TYPE REF TO data.
+    FIELD-SYMBOLS <ls_data> TYPE any.
+
+    DATA lt_content TYPE ty_content_t.
+    DATA ls_content TYPE ty_content_s.
+    DATA lr_ifdata TYPE REF TO ty_table_data_s.
+    FIELD-SYMBOLS <lt_table> TYPE ANY TABLE.
+
+    IF zif_abapgit_object~exists( ) = abap_false.
+      RETURN.
+    ENDIF.
+
+    TRY.
+
+        ASSIGN  lr_data TO <ls_data>.
+        IF NOT <ls_data> IS ASSIGNED.
+          RETURN.
+        ENDIF.
+
+        ls_icd_data_key-depl_scenario = ms_icd_data_key-depl_scenario.
+        ls_icd_data_key-ns = ms_icd_data_key-ns.
+        ls_icd_data_key-ifname = ms_icd_data_key-ifname.
+        ls_icd_data_key-ifver2 = ms_icd_data_key-ifver2.
+
+        TRY.
+            CALL METHOD mo_abapgit_util->('/AIF/IF_ABAPGIT_AIFC_UTIL~GET_IF_DATA')
+              EXPORTING
+                is_key    = ls_icd_data_key
+              RECEIVING
+                rt_ifdata = lt_ifdata.
+
+          CATCH cx_sy_dyn_call_error INTO lx_dyn_call_error.
+            zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                          ix_previous = lx_dyn_call_error ).
+        ENDTRY.
+
+        LOOP AT lt_ifdata REFERENCE INTO lr_ifdata.
+
+          UNASSIGN <lt_table>.
+          ASSIGN lr_ifdata->table_data->* TO <lt_table>.
+          IF <lt_table> IS NOT ASSIGNED.
+            CONTINUE.
+          ENDIF.
+
+          io_xml->add( iv_name = lr_ifdata->tabname
+                       ig_data = <lt_table> ).
+
+          ls_content-tabname = lr_ifdata->tabname.
+          APPEND ls_content TO lt_content.
+
+        ENDLOOP.
+
+        io_xml->add( iv_name = `Content_table`
+                     ig_data = lt_content ).
+
+      CATCH cx_root INTO lx_root.
+    ENDTRY.
```

Comment:
Looks suspicious. Above raised exceptions are catched silently here. Suggest to propagate them.